worker/peergrouper/desired.go

```diff
@@ -143,6 +155,18 @@ func (info *peerGroupInfo) initNewReplicaSet() map[string]*replicaset.Member {
 func desiredPeerGroup(info *peerGroupInfo) (map[string]*replicaset.Member, map[string]bool, error) {
 	logger.Debugf(info.getLogMessage())
 
+	peerChanges := peerGroupChanges{
+		isChanged:                   false,
+		toRemoveVote:                []string{},
+		toAddVote:                   []string{},
+		toKeepVoting:                []string{},
+		toKeepNonVoting:             []string{},
+		toKeepCreateNonVotingMember: []string{},
+		machineVoting:               map[string]bool{},
+		addrs:                       map[string]string{},
+		members:                     map[string]*replicaset.Member{},
```

Comment:
we probably only need to initialize the maps, as append(nil, str) works fine for adding entries to a []string. Its just that you can't v[key] = str if you're using a map.